### PR TITLE
Add default dest (0, 0) to Surface.blit

### PR DESCRIPTION
**Pull request by damusss**
Implements https://github.com/pygame-community/pygame-ce/issues/2935

I'm not sure if you want this to be added, personally I like it. I didn't discuss it because the implementation seemed pretty simple, so even if you don't want it it's not a big deal. 
The reasoning for it can be understood in the issue description.

I only modified blit and not blits/fblits, but I'm pretty sure I can add it to them too if you want, but for now I don't think it's a good idea.

### CODE REVIEW

**Comment by damusss** [comment]
I want to note that the failing check is related to numpy2 not this PR

File: buildconfig/stubs/pygame/surface.pyi [hunk 1]
```diff
@@ -69,7 +69,7 @@ class Surface:
     def blit(
         self,
         source: Surface,
-        dest: Union[Coordinate, RectValue],
+        dest: Optional[Union[Coordinate, RectValue]] = (0, 0),
```
Comment (ankith26) [inline buildconfig/stubs/pygame/surface.pyi:72]:
```suggestion
        dest: Union[Coordinate, RectValue] = (0, 0),
```
This is a common python typing pitfall...

`Optional` does not mean that an argument is optional, it means that it is allowed to be `None`, which isn't the case here

File: docs/reST/ref/surface.rst [hunk 2]
```diff
@@ -90,16 +90,16 @@
...
       Draws another Surface onto this Surface.
 
       **Parameters**
           - ``source``
               The ``Surface`` object to draw onto this ``Surface``.
               If it has transparency, transparent pixels will be ignored when blittting to an 8-bit ``Surface``.
-          - ``dest``
-              The ``source`` draw position onto this ``Surface``.
+          - ``dest`` *(optional)*
+              The ``source`` draw position onto this ``Surface``, defaults to (0, 0).
```
Comment (ankith26) [inline docs/reST/ref/surface.rst:102]: needs a `versionchanged` note

File: buildconfig/stubs/pygame/surface.pyi (hunk 1 above)
Comment (damusss) [reply buildconfig/stubs/pygame/surface.pyi]: Right, don't know how I let it happen! (fixed now). It's a bit weird that Optional doesn't mean optional, I'm glad the | operator exists in newer versions

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR :tada: 

This is an API change so I would like to have more eyes on this PR before we merge it in.

**Review by oddbookworm — APPROVED** [review]
LGTM!

**Review by itzpr3d4t0r — APPROVED** [review]
LGTM, should make it a bit faster as well in theory, and allows for shorter code.

**Pull request merged**